cli/src/git_util.rs

```diff
@@ -686,6 +703,36 @@ fn warn_if_branches_not_found(
     Ok(())
 }
 
+pub(crate) fn get_config_git_shell(command: &CommandHelper) -> Result<bool, ConfigGetError> {
+    command.settings().get_bool("git.shell").or_else(|e| {
+        if matches!(e, ConfigGetError::NotFound { .. }) {
+            Ok(false)
+        } else {
+            Err(e)
+        }
+    })
+}
+
+pub(crate) fn get_config_git_path(
+    command: &CommandHelper,
+) -> Result<Option<String>, ConfigGetError> {
+    if get_config_git_shell(command)? {
+        command
+            .settings()
+            .get_string("git.path")
```

Comment:
nit: maybe rename to `git.executable` or `git.program`? `git.path` sounds like a path to the `.git` directory.